plugins/builds/index.js

```diff
@@ -929,27 +1002,46 @@ exports.register = (server, options, next) => {
                         const parentBuild = await buildFactory.get(
                             build.parentBuilds[externalPipelineId].jobs[parentJobName]);
 
-                        newBuild = await createInternalBuild({
-                            jobFactory,
-                            buildFactory,
-                            eventFactory,
-                            pipelineId: externalEvent.pipelineId,
-                            jobName: externalJobName,
-                            username,
-                            scmContext,
-                            build: parentBuild, // this is the parentBuild for the next build
-                            baseBranch: event.baseBranch || null,
-                            parentBuilds: externalParentBuilds,
-                            parentBuildId: build.id,
-                            start: false
-                        });
+                        // if restart case, should create event
+                        if (isRestartCase) {
```

Comment:
`isRestartCase` build very much specific to external join build restart scenario. We need a name to reflect it